modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/README.md

```diff
@@ -0,0 +1,119 @@
+- [Search Pipelines](#search-pipelines)
+  - [Architecture](#architecture)
+  - [Search Processors](#search-processors)
+    - [Creating a Search Processor](#creating-a-search-processor)
+    - [Creating a Pipeline](#creating-a-search-pipeline)
+
+# Search Pipelines
+
+This README briefly covers the two types of search processors, explains how you can use them to create search pipelines, and walks through the creation of a new processor.
+
+## Architecture
+
+Search pipelines allow cluster operators to create and reuse [components](#search-processors) to transform search queries and results.
+
+With search pipelines, the operator can combine multiple [search processors](#search-processors) to create a transform which acts on the search request and/or search response.
+
+The primary benefit of search pipelines is that they live entirely inside OpenSearch and therefore require less development time. Further, search pipelines support standard APIs for accessing metrics and disaster recovery.
```

Comment:
I would say that the primary benefits are less about development time (versus the alternative of adding the logic to your own application) but rather:
1. Processors living in OpenSearch can be used by all calling applications (regardless of how they're implemented).
2. Processing (especially response processing) happens within the cluster (on the coordinator node) so larger results can be processed before returning to the calling application. Within the cluster, results are passed around using a more efficient (but very version-specific) binary protocol, so you can pass many results back to the coordinator, allow it to post-process (e.g. rerank or collapse) before truncating and sending a smaller result back to the client (using the less efficient, but more flexible JSON API).
3. Search processors can be distributed in plugins (or added to this module) to be shared with other OpenSearch users.

The other pre-existing "living entirely inside OpenSearch" option was writing your own `SearchPlugin`, but that's a lot more complicated as the `SearchPlugin` API is huge and (IMO) a bit overwhelming.

Reply:
Gotcha, I will reframe the wording. I was trying to consider the answer to "why use a search pipeline over a direct query?" as well as "why use a search pipeline over a plugin?" I will be more explicit in the cases for over a plugin that you listed.

I am not sure how else to provide insight into why over a query... The processors seem to mimic existing search query modifications, i.e. adding a Painless script into the call. Do you have a suggestion for how I could better answer the first question in the README?

Reply:
Oh! I forgot the other benefit that I wanted to list, which helps differentiate against modifying the query:

4. You can modify a search pipeline once on the OpenSearch cluster and have that take effect against all incoming queries (if they reference the pipeline directly or if they target an index configured to use the pipeline by default), without needing to modify and deploy all calling applications.

For example, the `FilterQueryRequestProcessor` could be used to exclude search results (by appending `NOT top_secret:true`) immediately (with several caveats -- this is not security advice), without needing to make a code change in the application layer and deploy that across your fleet.

Reply:
Oh great. I will clarify that. Thanks you for the better explanation :)